tlatools/org.lamport.tlatools/src/tlc2/debug/TLCSourceBreakpoint.java

```diff
@@ -27,14 +27,20 @@
 
 import org.eclipse.lsp4j.debug.SourceBreakpoint;
 
+import tla2sany.semantic.ModuleNode;
+import tla2sany.semantic.OpDefNode;
 import tla2sany.st.Location;
+import tlc2.tool.impl.SpecProcessor;
 
 public class TLCSourceBreakpoint extends SourceBreakpoint {
 
 	private final int hits;
 	private final Location location;
+	// Use Getter
+	public final OpDefNode condition;
```

Comment:
`condition` can be private if the implementation of the method `TLCStateStackFrame#matchesExpression` is moved into `TLCSourceBreakpoint` (`TLCStateStackFrame.matchesExpression` then delegates to `TLCSourceBreakpoint#matchesExpression).  Moreover, the impl no longer has to check `bp.getCondition() != null && !bp.getCondition().isEmpty()`.